src/object/obj_rpc.c

```diff
@@ -329,6 +329,62 @@ crt_proc_daos_iod_and_csum(crt_proc_t proc, crt_proc_op_t proc_op,
 	return rc;
 }
 
+static int crt_proc_daos_iom_t(crt_proc_t proc, daos_iom_t *map)
+{
+	crt_proc_op_t		 proc_op;
+	int			 i, rc;
+
+	rc = crt_proc_get_op(proc, &proc_op);
+	if (rc)
+		return rc;
+
+	rc = crt_proc_memcpy(proc, &map->iom_type, sizeof(map->iom_type));
+	if (rc != 0)
+		return -DER_HG;
+
+	rc = crt_proc_uint32_t(proc, &map->iom_nr);
+	if (rc != 0)
+		return -DER_HG;
+
+	rc = crt_proc_uint64_t(proc, &map->iom_size);
+	if (rc != 0)
+		return -DER_HG;
+
+	rc = crt_proc_daos_recx_t(proc, &map->iom_recx_lo);
+	if (rc != 0)
+		return -DER_HG;
+
+	rc = crt_proc_daos_recx_t(proc, &map->iom_recx_hi);
+	if (rc != 0)
+		return -DER_HG;
+
+	if (DECODING(proc_op)) {
+		D_ALLOC_ARRAY(map->iom_recxs, map->iom_nr);
```

Comment:
Just note that for client-side, after the IO completes the iom_recxs already be freeded. This means in fetch's complete CB need to dup it out, and in API header file need to comment user need to explict free it.

BTW, seems this patch did not implement all needed code, a few other code needed at client-side for above dup and ORF_CREATE_MAP flag setting. And actually I need to change the IOM related detailed handling in another EC patch, I can add those in my patch so just a synchronize with you here.

Reply:
Thank you for the note about the CB needing to dup it out. I will remember that. 

Indeed this patch by itself doesn't actually add the full feature for the map in the API. There is another ticket (DAOS-4459) that will finish the Fetch API portion. This change will provide the needed changes for checksums (DAOS-4460) to be able to understand holes on a fetch which is what I want to implement next. :-)

My thought was that by splitting it up this way I can get the checksums piece done without depending on the whole fetch API changes. Afterward, I, or someone else, can finish the API part.